storage/innobase/handler/ha_innodb.cc

```diff
@@ -5392,16 +5386,21 @@ static int innobase_init_files(dict_init_mode_t dict_init_mode,
     btr_search_enabled = old_btr_search_value;
   }
 
-  bool do_encrypt =
-      dict_detect_encryption(srv_is_upgrade_mode, upgrade_mysql_plugin_space);
+  bool do_encrypt = false;
+  bool ret = dict_detect_encryption_of_mysql_ibd(
+      dict_init_mode, upgrade_mysql_plugin_space,
+      dict_sys_t::s_dd_space_file_name, do_encrypt);
+  if (!ret) {
+    ib::error(ER_XB_MSG_2) << "Failed to determine if mysql.ibd is encrypted. "
```

Comment:
You are not supposed to re-use ER_ messages. Every message should introduce a unique id.